tests/tests.rs

```diff
@@ -1470,3 +1470,186 @@ fn test_base_directory() {
         ),
     );
 }
+
+/// --prune
+#[test]
+fn test_prune() {
+    let dirs = &["foo", "bar/zap/foo"];
+    let files = &[
+        "foo/a.foo",
+        "foo/b.bar",
+        "bar/zap/c.foo",
+        "bar/zap/foo/d.foo",
+    ];
+    let te = TestEnv::new(dirs, files);
+
+    // without pruning
+    te.assert_output(
+        &["foo"],
+        "bar/zap/foo/d.foo
+         bar/zap/c.foo
+         bar/zap/foo
+         foo/a.foo
+         foo",
+    );
+    // with pruning
+    te.assert_output(
+        &["--prune", "foo"],
+        "bar/zap/c.foo
+         bar/zap/foo
+         foo",
+    );
+
+    // with --full-path
```

Comment:
Thank you for adding all of these tests!

Maybe my initial comment came across wrong: is there any kind of interesting interplay between `--full-path` and `--prune`? If not, I'm not sure we need a test here.

Reply:
I didn't find anything too interesting. I suppose the only thing users would need to be aware of is if they use the `--full-path` flag  + the `--prune` option then if there are any directories within the full-path matching the pattern then the prune will take effect there, which might not have been pruned if a relative path had been used. But I don't feel that really warrants a test case IMO. If you are ok with it I can remove the --full-path test case from here.

Reply:
removed in commit 3